src/parse/asp/grammar.go

```diff
@@ -94,6 +94,21 @@ type OptimisedExpression struct {
 	Config string
 }
 
+// String is a string representation of OptimisedExpression
```

Comment:
Why do you need to string these? The AST types don't have a .String() for the most part and this type isn't meant to be used outside this package. You shouldn't need it for the LSP...